ml-agents/mlagents/trainers/torch/components/reward_providers/curiosity_reward_provider.py

```diff
@@ -106,38 +107,24 @@ def get_current_state(self, mini_batch: AgentBuffer) -> torch.Tensor:
         """
         Extracts the current state embedding from a mini_batch.
         """
-        n_vis = len(self._state_encoder.visual_processors)
-        hidden, _ = self._state_encoder.forward(
-            vec_inputs=[
-                ModelUtils.list_to_tensor(mini_batch["vector_obs"], dtype=torch.float)
-            ],
-            vis_inputs=[
-                ModelUtils.list_to_tensor(
-                    mini_batch["visual_obs%d" % i], dtype=torch.float
-                )
-                for i in range(n_vis)
-            ],
-        )
+        n_obs = len(self._state_encoder.encoders)
+        obs = ObsUtil.from_buffer(mini_batch, n_obs)
```

Comment:
Same comment about redefining `obs`. If mypy ever gets smart enough to understand the types it'll complain about this (currently it thinks they're both List[Any])